File: jshello/test/hellotest.js
var assert = require('assert');
var chai = require('chai');
var hello = require('../hello');

var expect = chai.expect;

describe('Hello', function() {
  describe('#message()', function() {
    it('should be "Hello, World!" when with World', function(){
	    let name = "World";
	    let expect = "Hello, World!";
	    let result = hello.helloMessage(name);
	    assert.equal(expect,result);
    });
    it('should be "Hello, Alice!" when with Alice', function(){
	    let name = "Alice";
	    let expect = "Hello, Alice!";
	    let result = hello.helloMessage(name);
	    assert.equal(expect,result);
    });
  });
});

File: py-hello/hellotest/hellotest.py
#!/usr/bin/env python

import inspect,os,sys,unittest

currentdir = os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe())))
parentdir = os.path.dirname(currentdir)

sys.path.insert(0,parentdir+"/hello")

from hello import helloMessage

class HelloTest(unittest.TestCase):
    def testHelloWorld(self):
        name = "World"
        expect = "Hello, World!"
        result = helloMessage(name)
        assert expect == result

    def testHelloAlice(self):
        name = "Alice"
        expect = "Hello, Alice!"
        result = helloMessage(name)
        assert expect == result

if __name__ == '__main__':
    unittest.main()
